test/litmus/type/date_time_test.exs

```diff
@@ -0,0 +1,66 @@
+defmodule Litmus.Type.DateTimeTest do
+  use ExUnit.Case, async: true
+  doctest Litmus.Type.DateTime
+
+  alias Litmus.Type
+
+  describe "validate_field/3" do
+    test "converts a string to a datetime" do
+      field = "start_date"
+      data = %{"start_date" => "1990-05-01T06:32:00Z"}
+
+      {:ok, expected_date_time, _} = DateTime.from_iso8601(data["start_date"])
+      expected_data = %{"start_date" => expected_date_time}
+
+      type = %Type.DateTime{
+        required: true
+      }
+
+      assert Type.DateTime.validate_field(type, field, data) == {:ok, expected_data}
+    end
+
+    test "does not alter nil" do
+      field = "start_date"
+      data = %{"start_date" => nil}
+
+      type = %Type.DateTime{}
+
+      assert Type.DateTime.validate_field(type, field, data) == {:ok, data}
+    end
```

Comment:
Why are we allowing `nil`?

Reply:
We are allowing `nil` for if the parameter is optional, but not if it is required. I think this became the most straightforward interpretation of how this library should work so far. A `nil` attribute is  like not passing the attribute at all, sort of. Happy to discuss more if you think this is wrong.